### PR TITLE
Fixes #18072 - removed load_subnets and load_subnet_data methods from dhcp provider interface

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by dmitri-d** [comment]
@timogoebel: another PR to fix issues discovered when you were working on support for bonded interfaces.

File: modules/dhcp_libvirt/dhcp_libvirt_main.rb [hunk 1]
```diff
@@ -1,83 +1,14 @@
...
-require 'ipaddr'
-
 module Proxy::DHCP::Libvirt
   class Provider < ::Proxy::DHCP::Server
     attr_reader :libvirt_network, :network
 
     def initialize(network, libvirt_network_impl, subnet_service)
       @network = network
       @libvirt_network = libvirt_network_impl
+      @subnet_service_initialized = false
```
Comment (timogoebel) [inline modules/dhcp_libvirt/dhcp_libvirt_main.rb]: Where is this used?

**Review by timogoebel — COMMENTED** [review]
@witlessbird : Looks good so far. Just one question inline.

File: modules/dhcp_libvirt/dhcp_libvirt_main.rb (hunk 1 above)
Comment (dmitri-d) [reply modules/dhcp_libvirt/dhcp_libvirt_main.rb]: It's no longer being used; removed.

**Comment by dmitri-d** [comment]
- removed unused instance variable @subnet_service_initialized in modules/dhcp_libvirt/dhcp_libvirt_main.rb.

**Review by timogoebel — APPROVED** [review]
@witlessbird : Thanks! Looks good to me, improved the API a lot and works great with isc-dhcp. 👍

**Pull request merged**